plugin/src/main/java/hudson/plugins/swarm/SwarmSlave.java

```diff
@@ -77,10 +77,15 @@ public boolean isInstantiable() {
 
         @Override
         public void afterDisconnect(SlaveComputer computer, TaskListener listener) {
-            try {
-                Jenkins.getInstance().removeNode(computer.getNode());
-            } catch (IOException e) {
-                e.printStackTrace(listener.error(e.getMessage()));
+            final Slave node = computer.getNode();
+            if (node != null) {
+                try {
+                    Jenkins.getInstance().removeNode(node);
```

Comment:
Minor: This method call also can return null. It should not happens in the handler